prow/tekton.md

```diff
@@ -0,0 +1,142 @@
+you also need to install and setup Tekton Pipelines itself.
+
+[Prow is currently compatible with versions 0.2.0 - 0.3.1 of Tekton Pipelines](https://github.com/kubernetes/test-infra/issues/13948)
+so you must install one of those versions, e.g.:
+
+```bash
+kubectl apply --filename  https://storage.googleapis.com/tekton-releases/previous/v0.3.1/release.yaml
+```
+
+_The [configuration used above](#pipeline-and-crier-controllers)
+for the Prow Pipeline controller assumes that Tekton Pipelines is
+installed into its default namespace (`tekton-pipelines`)._
+
+## Configuring a `tekton-pipeline` job
+
+Once [you have everything setup](#setting-up-pipelines-with-prow) you can
+configure Prow jobs to run Pipelines.
+
+1. [Create and apply `Pipelines` and
+   `Tasks`](#create-and-apply-pipelines-and-tasks)
+2. [Configure Prow with `ProwJobs` that use those Pipelines and Tasks](#add-tekton-pipeline-prowjobws)
+3. Then when the `ProwJob` is triggered:
+  a. Prow (via [the Prow pipeline controller](#pipeline-and-crier-controllers))
+     will create [a PipelineRun](https://github.com/tektoncd/pipeline/blob/master/docs/pipelineruns.md)
+  b. When the `PipelineRun` completes, the Prow Pipeline controller will update
+     the `ProwJob`
+  c. Finally, [crier](#pipeline-and-crier-controllers) will see the update to
+     the `ProwJob` and update `GitHub` with the results
+
+### Create and apply `Pipelines` and `Tasks`
+
+Before anything can be executed, Tekton expects
+[`Pipelines`](https://github.com/tektoncd/pipeline/blob/master/docs/pipelines.md)
+and the [`Tasks`](https://github.com/tektoncd/pipeline/blob/master/docs/tasks.md)
+that the `Pipelines` reference to exist.
+
+Since [Prow is only compatible with versions 0.2.0 - 0.3.1 of Tekton Pipelines](https://github.com/kubernetes/test-infra/issues/13948)
+the below docs are pinned to v0.3.1 of Tekton Pipelines:
+
+* See [the Tekton tutorial](https://github.com/tektoncd/pipeline/blob/v0.3.1/docs/tutorial.md)
+  for an overview
+* See [the Tekton Pipelines docs](https://github.com/tektoncd/pipeline/tree/v0.3.1/docs#tekton-pipelines)
+  for reference docs
+* See [the Tekton catalog](https://github.com/tektoncd/catalog)
+  for examples of `Tasks` you might want to use.
+
+### Add `tekton-pipeline` ProwJobs
+
+To configure a `ProwJob` to use Tekton Pipelines, include a `pipelineRunSpec`,
+for example:
+
+```yaml
+presubmits:
+  bobcatfish/catservice:
+  - name: do-the-pipeline
+    agent: tekton-pipeline # Use the Prow Tekton controller!
+    rerun_command: "/run do-the-pipeline"
+    trigger: "(?m)^/run do-the-pipeline,?(\\s+|$)"
+    always_run: true
+    skip_report: false
+    pipeline_run_spec: # Template for creation of the `PipelineRun`
+      trigger: # Required by versions 0.2.0 - v0.3.1 of Tekton Pipelines
+        type: manual
+      pipelineRef:
+        name: special-pipeline # Use the Tekton Pipeline called `special-pipeline`
```

Comment:
I think the `serviceAccount` needs also to be defined to use the Tekton's service account instead of `default`. That's the service account with the GitHub token linked to it.

Reply:
>  It should be the Tekton's service account with the GitHub token linked to it.

Maybe you can explain more about what this service account would be doing? You can [optionally provide a serviceAccount to a PipelineRun](https://github.com/tektoncd/pipeline/blob/master/docs/pipelineruns.md#service-account) but that just makes it available to executing Tasks if they need it (e.g. to push to an image registry).

Re the GitHub token, if the PipelineRun was trying to push something to GitHub for example, if you wanted [to use the PullRequest resource](https://github.com/tektoncd/pipeline/blob/master/docs/resources.md#pull-request-resource) and push to GitHub from the Pipeline itself, you'd need to provide a secret in the `PipelineResource`

TL;DR it depends on what the PipelineRun is doing - I can add a link to the PipelineRun docs and try to make that more clear here.

> I am not sure that service account is used by default by the Tekton controller.

The controller uses the [` tekton-pipelines-controller` service account](https://github.com/tektoncd/pipeline/blob/master/config/200-serviceaccount.yaml) by default (https://github.com/tektoncd/pipeline/blob/master/config/controller.yaml#L28) - but pods that execute as part of a PipelineRun would use the default service account of the namespace by default, unless `serviceAccount` or `serviceAccounts` are specified

Reply:
I've added a link to the PipelineRun docs right below this and mentioned ServiceAccounts!

Reply:
I mean this service account https://github.com/tektoncd/pipeline/blob/master/docs/auth.md#basic-authentication-git, but pointing to the doc is enough if someone needs to configure it. Thanks for adding that!